src/main/java/codegym/tequila/fisioapp/service/impl/TherapyServiceImpl.java

```diff
@@ -0,0 +1,128 @@
+            throw new IllegalArgumentException("Therapy must have a description");
+        }
+
+        Therapy therapy = new Therapy();
+
+        therapy.setId(UUID.randomUUID().toString());
+        therapy.setName(therapyDto.getName());
+        therapy.setDescription(therapyDto.getDescription());
+        therapy.setActive(true);
+
+        return convertTherapyToDto(therapyRepository.save(therapy));
+    }
+
+    @Transactional
+    @Override
+    public TherapyDto updateTherapy(String therapyId, TherapyDto therapyDto) {
+
+        Therapy therapy = therapyRepository.findById(therapyId)
+                .orElseThrow(() -> new NoSuchElementException("Therapy " + therapyId + " not found"));
+
+        if (StringUtils.hasLength(therapyDto.getName())) {
+            therapy.setName(therapyDto.getName());
+        }
+
+        if (StringUtils.hasLength(therapyDto.getDescription())) {
+            therapy.setDescription(therapyDto.getDescription());
+        }
+
+        return convertTherapyToDto(therapyRepository.save(therapy));
+    }
+
+    @Override
+    public void deactivateTherapy(String therapyId) {
+        Therapy therapy = therapyRepository.findById(therapyId)
+                .orElseThrow(() -> new NoSuchElementException("Therapy " + therapyId + " not found"));
+
+        therapy.setActive(false);
+        therapyRepository.save(therapy);
+    }
+
+    @Override
+    public void activateTherapy(String therapyId) {
+        Therapy therapy = therapyRepository.findById(therapyId)
+                .orElseThrow(() -> new NoSuchElementException("Therapy " + therapyId + " not found"));
+
+        therapy.setActive(true);
+        therapyRepository.save(therapy);
+    }
+
+
+    @Override
+    public TherapyDto getTherapy(String therapyId) {
+        Therapy therapy = therapyRepository
+                .findById(therapyId).orElseThrow(() -> new NoSuchElementException("Therapy " + therapyId + " not found"));
+
+        return convertTherapyToDto(therapy);
+    }
+
+    @Override
+    public List<TherapyDto> getTherapies(Integer pageSize, Integer pageIndex, Boolean all, Boolean inactive) {
+
+        pageSize = pageSize == null ? 10 : pageSize;
+        pageIndex = pageIndex == null ? 0 : pageIndex;
+        all = all != null && all;
+        inactive = inactive != null && inactive;
+
+        Pageable pageable = PageRequest.of(pageIndex, pageSize);
+
+        if (all) {
+            return therapyRepository.findAll(pageable).stream()
+                    .map(TherapyServiceImpl::convertTherapyToDto)
+                    .collect(Collectors.toList());
+        }
+
+        if (inactive) {
+            return therapyRepository.findAllByActive(false, pageable).stream()
+                    .map(TherapyServiceImpl::convertTherapyToDto)
+                    .collect(Collectors.toList());
+        } else {
+            return therapyRepository.findAllByActive(true, pageable).stream()
```

Comment:
No need to change if you prefer this way but I'd prefer to get the list of Therapies from the repository and at the end one single return statement with the conversion: 
```
return therapiesList.stream()
                    .map(TherapyServiceImpl::convertTherapyToDto)
                    .collect(Collectors.toList());
```
This way you avoid the duplicated code.